unumpy/_multimethods.py

```diff
@@ -1145,6 +1343,78 @@ def searchsorted(a, v, side="left", sorter=None):
     return (a,)
 
 
+def _take_default(a, indices, axis=None, out=None, mode="raise"):
+    if axis is None:
+        a = ravel(a)
+        axis = 0
+
+    nd = ndim(a)
+
+    axis = _normalize_axis(nd, axis)
+
+    if mode not in {"raise", "wrap", "clip"}:
+        raise TypeError("Clipmode not understood.")
+
+    if mode == "wrap":
+        indices = mod(indices, a.shape[axis])
+    elif mode == "clip":
+        indices = clip(indices, 0, a.shape[axis] - 1)
+
+    slices = [slice(None)] * nd
+    slices[axis] = indices
+    slices = tuple(slices)
+
+    res = a[slices]
+
+    if out is None:
+        return res
+    else:
+        copyto(out, res)
+
+
+@create_numpy(_first2argreplacer, default=_take_default)
+@all_of_type(ndarray)
+def take(a, indices, axis=None, out=None, mode="raise"):
+    return (a, indices, mark_non_coercible(out))
+
+
+@create_numpy(_first2argreplacer)
+@all_of_type(ndarray)
+def take_along_axis(arr, indices, axis):
+    return (arr, indices)
+
+
+def _choose_default(a, choices, out=None, mode="raise"):
+    if mode == "raise":
+        if any(a < 0) or any(a >= len(choices)):
+            raise ValueError("Invalid entry in choice array.")
+    elif mode == "wrap":
+        a = mod(a, len(choices))
+    elif mode == "clip":
+        a = clip(a, 0, len(choices) - 1)
+    else:
+        raise TypeError("Clipmode not understood.")
+
+    choices = [asarray(choice) for choice in choices]
+
+    a, *choices = broadcast_arrays(a, *choices)
+
+    merged_array = empty(a.shape, dtype=int)
+    for i, c in enumerate(choices):
+        merged_array = where(a == i, c, merged_array)
+
+    if out is None:
+        return merged_array
+    else:
+        copyto(out, merged_array)
+
+
+@create_numpy(_self_out_argreplacer, default=_choose_default)
+@all_of_type(ndarray)
+def choose(a, choices, out=None, mode="raise"):
```

Comment:
Dispatch each item in `choices` and/or dispatch `choices`?

Reply:
I think it would have to be both, so maybe something like:
```python
if isinstance(choices, (tuple, list)):
   return (a, mark_non_coercible(out)) + tuple(choices)
else:
  return (a, choices, mark_non_coercible(out))
```